### ISSUE TITLE
Ignore no-object-as-default-parameter for objects with required parameters

### ISSUE THREAD

**Issue by ST-DDT** [sindresorhus/eslint-plugin-unicorn#2199]
Sometimes a method has a required set of option parameters. e.g. min and max.

````ts
function valueBetween(range: {min: number, max: number} = { min: 0, max: 10 }) {} ✔️
````

In this case it should be OK to use object default parameters because both min and max are required.
Or to be more precise, it should be ok if defined default properties === required properties.
It is not OK if any optional property is provided.

Yes, it it possible for plain JS users to fall through, but they are also likely to fail with the types in general or cause conflicts due to mismatching boundaries:

````js
valueBetween({min: 'a', max: 10}); ❌
valueBetween({min: 100}); ❌
````

Otherwise it is impossible to implement the method parameter default in a type safe way.

````ts
function valueBetween(range: {min: number, max: number} = {} ❌) {
    const { min = 0, max = 10 } = options;
}
````

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-object-as-default-parameter.md

````ts
function valueBetween(range: {min: number, max: number} = { min: 0, max: 10 }) {} :check ✔️
function valueBetween(range: {min: number, max: number, step: number} = { min: 0, max: 10 }) {} :check ❌
function valueBetween(range: {min: number, max: number, step?: number} = { min: 0, max: 10, step: 1 }) {} :check ❌
````

https://github.com/faker-js/faker/blob/fe3fb5dd3b8949d952aa8eb3830f58fc9cc4adcc/src/modules/lorem/index.ts#L377

**Comment by sindresorhus** [comment]
I don't think we should support this.

The rule is intentionally syntax based: a non-empty object literal as a parameter default is risky because a caller can pass a partial object and replace the whole default. TypeScript can reject that in some codebases, but the runtime behavior is still the same, and Unicorn should not assume every caller is typed correctly.

Supporting the proposed exception would require proving that the default object's keys exactly match the required keys of the parameter type. That quickly becomes type aware and inconsistent: an inline type literal might be detectable, while an equivalent type alias, interface, import, union, intersection, or utility type would not be without full type resolution.

For APIs where the object is intentionally atomic and callers must provide all fields, a local disable is clearer than making this rule much more complex. The general recommendation remains to default to `{}` and destructure individual fields when partial options are supported.
